src/jeco/core/algorithm/moge/GrammaticalEvolution.java

```diff
@@ -193,24 +192,32 @@ public void step() {
   }
   
   protected void collectStatistics() {
-	  Double currentBest = this.population.get(0).getObjectives().get(0);
+	  Solution<Variable<Integer>> currentBest = this.population.get(0);
+	  Comparator<Solution<Variable<Integer>>> comparator = this.dominance;
 	  
-	  if (currentBest < this.absoluteBest) {
-		  this.absoluteBestObjetives.add(currentBest);
+	  if (this.absoluteBest == null || comparator.compare(currentBest, this.absoluteBest) < 0) {
+		  this.absoluteBestObjetives.add(currentBest.getObjectives());
 		  this.absoluteBest = currentBest;
 	  }
 	  else
-		  this.absoluteBestObjetives.add(this.absoluteBest);
+		  this.absoluteBestObjetives.add(this.absoluteBest.getObjectives());
 	  
-	  Double sum = 0.0;
+	  ArrayList<Double> avg = new ArrayList<>(population.get(0).getObjectives().size());
+	  for (int i = 0; i < population.get(0).getObjectives().size(); i++) {
```

Comment:
Este for es equivalente a `Arrays.fill(avg, 0.0)`. Por si te interesa para la próxima vez.